### PR TITLE
Change algorithm that breaks java source code into separate words.

### PR DESCRIPTION
**Pull request by andriimatsokha**
Algorithm that breaks java source code into words is based on finite state machine. It breaks words accordingly to Java Code Convention.
Remove unused class SpellCheckerUtil.

### CODE REVIEW

File: src/main/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/WordFinderAutomaton.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package com.epam.sonarqube.spellcheck.plugin.spellcheck.automaton;
+
+public interface WordFinderAutomaton {
+
+    /**
+     * Initialize automaton with states and rules for parsing source code
+     */
+    public void init();
```
Comment (webdizz) [inline src/main/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/WordFinderAutomaton.java]: public is unnecessary for interface

File: src/main/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/WordFinderAutomaton.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    
+    /**
+     * Return start position of word, which was found by previous call of
+     * searchNextWord(...)
+     * 
+     * @return int start position of found word
+     */
+    public abstract int getWordStart();
```
Comment (webdizz) [inline src/main/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/WordFinderAutomaton.java]: abstract is unnecessary for interface

File: src/main/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/states/State.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    private String name;
+
+    /* Map of all transitions */
+    private Map<Character, State> transitions;
+    /* Default transition */
+    private State elseState;
+    
+    private StateCallback callback;
+    
+    protected boolean isFinalState = false;
```
Comment (webdizz) [inline src/main/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/states/State.java]: fields should be sorted according to visibility public, protected, private etc

File: src/test/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/states/FinalStateTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    
+    private StateEvent stateEvent;
+
+    @Before
+    public void init() {
+        stateEvent = new StateEvent(automaton);
+    }
+    
+    @Test
+    public void testCallbackCallsBehavior() throws Exception {
```
Comment (webdizz) [inline src/test/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/states/FinalStateTest.java]: test should start from "should" and continue as a narrative what it should check
for example shouldVerifyEventPassed and similar

File: src/test/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/JavaCodeConventionEnglishAutomatonTest.java [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+    private JavaCodeConventionEnglishAutomaton automaton;
+
+    @Before
+    public void init() {
+        automaton = new JavaCodeConventionEnglishAutomaton();
+        automaton.init();
+    }
+
+    @Test
+    public void testInit() throws Exception {
```
Comment (webdizz) [inline src/test/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/JavaCodeConventionEnglishAutomatonTest.java]: test method should be called "should" + narrative

File: src/test/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/JavaCodeConventionEnglishAutomatonTest.java [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+        assertEquals(-1, automaton.getWordStart());
+        assertEquals(0, automaton.getWordEnd());
+    }
+
+    @Test
+    public void testHasNextWordNoNextWord() throws Exception {
+        assertFalse(automaton.hasNextWord());
```
Comment (webdizz) [inline src/test/java/com/epam/sonarqube/spellcheck/plugin/spellcheck/automaton/JavaCodeConventionEnglishAutomatonTest.java]: use hamcrest assertions or provide a failure reason message

**Comment by webdizz** [comment]
Please pull the latest changes so that your PR is up-to-date

**Comment by andriimatsokha** [comment]
Fixed all issues mentioned in comments and pulled latest changes.

**Pull request merged**